mars/dataframe/initializer.py

```diff
@@ -19,17 +19,17 @@
 
 from ..tensor.core import TENSOR_TYPE
 from .core import DATAFRAME_TYPE, DataFrame as _Frame
-from .expressions.datasource.dataframe import from_pandas
+from .expressions.datasource.dataframe import from_pandas, from_tensor
 
 
 class DataFrame(_Frame):
     def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False,
                  chunk_size=None, gpu=None, sparse=None):
         if isinstance(data, TENSOR_TYPE):
-            raise NotImplementedError('Not support create DataFrame from tensor')
-        if isinstance(data, DATAFRAME_TYPE):
-            raise NotImplementedError('Not support yet')
-
-        pdf = pd.DataFrame(data, index=index, columns=columns, dtype=dtype, copy=copy)
-        df = from_pandas(pdf, chunk_size=chunk_size, gpu=gpu, sparse=sparse)
+            df = from_tensor(data)
+        elif isinstance(data, DATAFRAME_TYPE):
+            df = from_pandas(data, chunk_size=chunk_size, gpu=gpu, sparse=sparse)
```

Comment:
`DATAFRAME_TYPE` means Mars DataFrame type, `from_pandas` is used for creating from pandas.  Here we just return the `data` if it is a type of Mars DataFrame